core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -354,10 +355,12 @@ class KafkaApis(val requestChannel: RequestChannel,
             offsetMetadata = OffsetMetadata(partitionData.offset, metadata),
             commitTimestamp = currentTimestamp,
             expireTimestamp = {
-              if (partitionData.timestamp == OffsetCommitRequest.DEFAULT_TIMESTAMP)
-                defaultExpireTimestamp
+              if (header.apiVersion >= 3)
+                None
+              else if (partitionData.timestamp == OffsetCommitRequest.DEFAULT_TIMESTAMP)
+                Some(defaultExpireTimestamp)
```

Comment:
Can we use `None` here? I thought the idea in the KIP was to treat older version of the offset commit which have the default timestamp set under the new expiration logic.